packages/react-jsx-highcharts/src/components/Axis/Axis.js

```diff
@@ -83,7 +61,26 @@ class Axis extends Component {
   }
 
   render () {
-    if (!this.axis) return null;
+    if (!this.axis) {
+      const { id, dynamicAxis, isX, getChart } = this.props;
```

Comment:
Interesting, good spot on this! But having this logic in render makes me feel uneasy.

I know it is exactly the same meaning but could we extract this out into...

```jsx
render () {
  if (!this.axis) this.initialRender() // line 65-83 in this function
  // As before
}

Reply:
I agree that setting instance variables in render() seems like an anti-pattern. Unfortunately, there is no better lifecycle method to do that.

I extracted methods: createAxis() and createSeries(). Those names felt more explicit.